execute/outcome.go

```diff
@@ -199,11 +201,12 @@ func (p *Plugin) getFilterOutcome(
 		return exectypes.Outcome{}, fmt.Errorf("unable to select report: %w", err)
 	}
 
-	execReport := cciptypes.ExecutePluginReport{
-		ChainReports: outcomeReports,
+	// Collapse the commit reports.
+	var mergedData []exectypes.CommitData
+	for _, commitData := range selectedCommitReports {
+		mergedData = append(mergedData, commitData...)
 	}
```

Comment:
This is kinda weird, so `selectedCommitReports` is not already flattened? How did the types even work out?